src/test/java/build/buildfarm/instance/memory/ByteStringStreamSourceTest.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm.instance.memory;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.io.IOException;
+import java.io.InputStream;
+
+@RunWith(JUnit4.class)
+public class ByteStringStreamSourceTest {
+  private class CountingRunnable implements Runnable {
+    int runCount = 0;
+
+    @Override
+    public void run() {
+      runCount++;
+    }
+  };
+
+  @Test
+  public void closeCallsOnClose() throws IOException {
+    CountingRunnable counter = new CountingRunnable();
+    ByteStringStreamSource source = new ByteStringStreamSource(counter);
```

Comment:
Note you could also do it with mocks -- verifying how many times was something called is a very typical mock use-case. But it's simple enough either way, so up to you.